src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -0,0 +1,1061 @@
+pub struct SharedFsInfo {
+    /// Shared file system type:
+    /// - virtio-fs (default)
+    /// - virtio-9p`
+    pub shared_fs: Option<String>,
+
+    /// Path to vhost-user-fs daemon.
+    #[serde(default)]
+    pub virtio_fs_daemon: String,
+
+    /// List of valid annotations values for the virtiofs daemon
+    /// The default if not set is empty (all annotations rejected.)
+    #[serde(default)]
+    pub valid_virtio_fs_daemon_paths: Vec<String>,
+
+    /// Extra args for virtiofsd daemon
+    ///
+    /// Format example:
+    ///    ["-o", "arg1=xxx,arg2", "-o", "hello world", "--arg3=yyy"]
+    ///
+    ///  see `virtiofsd -h` for possible options.
+    #[serde(default)]
+    pub virtio_fs_extra_args: Vec<String>,
+
+    /// Cache mode:
+    /// - none: Metadata, data, and pathname lookup are not cached in guest. They are always
+    ///   fetched from host and any changes are immediately pushed to host.
+    /// - auto: Metadata and pathname lookup cache expires after a configured amount of time
+    ///   (default is 1 second). Data is cached while the file is open (close to open consistency).
+    /// - always: Metadata, data, and pathname lookup are cached in guest and never expire.
+    #[serde(default)]
+    pub virtio_fs_cache: String,
+
+    /// Default size of DAX cache in MiB
+    #[serde(default)]
+    pub virtio_fs_cache_size: u32,
+
+    /// Enable virtio-fs DAX window if true.
+    #[serde(default)]
+    pub virtio_fs_is_dax: bool,
+
+    /// This is the msize used for 9p shares. It is the number of bytes used for 9p packet payload.
+    #[serde(default)]
+    pub msize_9p: u32,
+}
+
+impl SharedFsInfo {
+    /// Adjust the configuration information after loading from configuration file.
+    pub fn adjust_configuration(&mut self) -> Result<()> {
+        if self.shared_fs.as_deref() == Some("") {
+            self.shared_fs = Some(default::DEFAULT_SHARED_FS_TYPE.to_string());
+        }
+        match self.shared_fs.as_deref() {
+            Some(VIRTIO_FS) => self.adjust_virtio_fs(false)?,
+            Some(VIRTIO_FS_INLINE) => self.adjust_virtio_fs(true)?,
+            Some(VIRTIO_9P) => {
+                if self.msize_9p == 0 {
+                    self.msize_9p = default::DEFAULT_SHARED_9PFS_SIZE;
+                }
+            }
+            _ => {}
+        }
+
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    pub fn validate(&self) -> Result<()> {
+        match self.shared_fs.as_deref() {
+            None => Ok(()),
+            Some(VIRTIO_FS) => self.validate_virtio_fs(false),
+            Some(VIRTIO_FS_INLINE) => self.validate_virtio_fs(true),
+            Some(VIRTIO_9P) => {
+                if self.msize_9p < default::MIN_SHARED_9PFS_SIZE
+                    || self.msize_9p > default::MAX_SHARED_9PFS_SIZE
+                {
+                    return Err(eother!(
+                        "Invalid 9p configuration msize 0x{:x}",
```

Comment:
I think printing min & max here would be helpful.

Reply:
fixed